lua/nvim-treesitter/highlight.lua

```diff
@@ -16,13 +40,37 @@ function M.attach(bufnr, ft)
   M.highlighters[buf] = ts.TSHighlighter.new(query, buf, ft)
 end
 
-function M.detach(bufnr)
-  local buf = bufnr or api.nvim_get_current_buf()
-  if M.highlighters[buf] then
-    M.highlighters[buf]:set_query("")
-    M.highlighters[buf] = nil
+local function enable_hl()
+  M.setup()
+  for _, buf in pairs(api.nvim_list_bufs()) do
+    M.attach(buf)
+  end
+end
+
+function M.setup()
+  for _, ft in pairs(configs.available_parsers()) do
+    if utils.has_parser(ft)
+      and queries.get_query(ft, 'highlights')
+      and not M.registered_autocmd[ft] then
+      local cmd = "lua require'nvim-treesitter.highlight'.attach()"
+      local autocmd = string.format('autocmd Filetype %s %s', ft, cmd)
```

Comment:
It is recommended to include the autocommands in a `augroup` so it should be like this 
```suggestion
      local autocmd = string.format('autocmd NvimTreesitter Filetype %s %s', ft, cmd)
```